### PR TITLE
[ButtonBar] Expose a rectForItem:inCoordinateSpace: API.

### LINKED ISSUE #7248: [NavigationBar] Expose a mechanism for getting the frame of a button
This was filed as an internal issue. If you are a Googler, please visit [b/131155047](http://b/131155047) for more details.

---
#### Internal data
- Associated internal bug: [b/131155047](http://b/131155047)

### PR DESCRIPTION
**Pull request by jverkoey**
This API will allow clients to present overlay views that point to views in an MDCButtonBar.

Expected usage:

```swift
@objc func didTapButton(_ item: UIBarButtonItem) {
  let rect = buttonBar.rect(for: item, in: view)
  // Present an overlay pointing at `rect`
}
```

Closes https://github.com/material-components/material-components-ios/issues/7248

### CODE REVIEW

File: components/ButtonBar/src/MDCButtonBar.h [hunk 1]
```diff
@@ -98,6 +98,17 @@ IB_DESIGNABLE
...
 
+/**
+ Returns the rect of the item's view within the coordinate space of @c view.
+
+ If the provided item is not contained in @c items, then the behavior is undefined.
+
+ @param item The item within @c items whose rect should be computed.
+ @param coordinateSpace The coordinate space the returned rect should be in relation to.
+ */
+- (CGRect)rectForItem:(nonnull UIBarButtonItem *)item
```
Comment (codeman7) [inline components/ButtonBar/src/MDCButtonBar.h:109]: Should this be `boundsForItem:...`

File: components/ButtonBar/tests/unit/ButtonBarRectForItemTests.swift [hunk 2]
```diff
@@ -0,0 +1,108 @@
+// Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (codeman7) [inline components/ButtonBar/tests/unit/ButtonBarRectForItemTests.swift]: Tip: Please update the copyright.

File: components/ButtonBar/tests/unit/ButtonBarRectForItemTests.swift [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import MaterialComponents.MaterialButtonBar
+import MaterialComponents.MaterialButtons
+
+// Tests for Button Bar's rectForItemAtIndex:inCoordinateSpace: API.
+class ButtonBarRectForItemTests: XCTestCase {
+
+  var buttonBar: MDCButtonBar!
+
+  override func setUp() {
+    buttonBar = MDCButtonBar()
```
Comment (codeman7) [inline components/ButtonBar/tests/unit/ButtonBarRectForItemTests.swift:27]: Tip: Please call super and add a teardown.

**Review by codeman7 — APPROVED** [review]

File: components/ButtonBar/src/MDCButtonBar.h (hunk 1 above)
Comment (jverkoey) [reply components/ButtonBar/src/MDCButtonBar.h:109]: This naming matches UIKit's convention in the `convertRect:toCoordinateSpace:` family of APIs that this API is modeled after.

**Pull request merged**